src/test/resources/com/jetbrains/snakecharm/cucumber/completion/keywords.feature

```diff
@@ -254,61 +190,67 @@ Feature: Completion for snakemake keyword-like things
       | subworkflow |
       | rule        |
 
-  Scenario Outline: Complete at checkpoint level
+  Scenario Outline: Complete at rule/checkpoint level
     Given a snakemake project
     Given I open a file "foo.smk" with text
-    """
-    checkpoint NAME:
-      <str>
-    """
+      """
+      <rule_like> NAME:
+        <str>
+      """
     When I put the caret after <str>
     Then I invoke autocompletion popup, select "<result>" lookup item and see a text:
-    """
-    checkpoint NAME:
-      <result>: 
-    """
+      """
+      <rule_like> NAME:
+        <result>: 
+      """
     Examples:
-      | str | result               |
-      | inp | input                |
-      | out | output               |
-      | par | params               |
-      | lo  | log                  |
-      | re  | resources            |
-      | be  | benchmark            |
-      | ve  | version              |
-      | me  | message              |
-      | th  | threads              |
-      | si  | singularity          |
-      | pr  | priority             |
-      | wi  | wildcard_constraints |
-      | gr  | group                |
-      | sh  | shadow               |
-      | co  | conda                |
-      | cw  | cwl                  |
-      | sc  | script               |
-      | sh  | shell                |
-      | run | run                  |
-      | wr  | wrapper              |
+      | rule_like  | str    | result               |
+      | rule       | inp    | input                |
+      | checkpoint | inp    | input                |
+      | rule       | out    | output               |
+      | checkpoint | out    | output               |
+      | rule       | par    | params               |
+      | checkpoint | lo     | log                  |
+      | rule       | re     | resources            |
```

Comment:
Cucumber doesn't allow to multiply params values, e.g. generate 2 (rule_like) x 23 (str & result) combinations. But here I don't see much sense in testing half features for rule and half for checkpoints. So I see two approaches 1) test full list for 'rule' and check several cases for 'checkpoint' 2) test everything for rule and checkpoint.

Reply:
I don't have any ideas how to avoid any copy paste here =/